torchvision/transforms/functional.py

```diff
@@ -469,6 +470,16 @@ def center_crop(img: Tensor, output_size: List[int]) -> Tensor:
     image_width, image_height = _get_image_size(img)
     crop_height, crop_width = output_size
 
+    if crop_width > image_width or crop_height > image_height:
+        padding_ltrb = [
+            (crop_width - image_width) // 2 if crop_width > image_width else 0,
+            (crop_height - image_height) // 2 if crop_height > image_height else 0,
+            (crop_width - image_width + 1) // 2 if crop_width > image_width else 0,
+            (crop_height - image_height + 1) // 2 if crop_height > image_height else 0,
+        ]
+        img = pad(img, padding_ltrb, fill=0)  # PIL uses fill value 0
+        image_width, image_height = _get_image_size(img)
```

Comment:
I think we can also check if we only padded the image and thus no need to crop, right ?

Reply:
Yeah, missed that(sorry :facepalm: ). I will get back on this a little later. The fix would require also adding test cases since currently cropsize > imagesize along both dimensions. As of now, all 4 test samples only require padding, so crop method will never be called.